fractalis/analytics/tasks/boxplot/main.py

```diff
@@ -25,7 +26,8 @@ def main(self,
              categories: List[pd.DataFrame],
              id_filter: List[T],
              transformation: str,
-             subsets: List[List[T]]) -> dict:
+             subsets: List[List[T]],
+             subset_labels: List[str]) -> dict:
```

Comment:
You could add a default here, e.g., `subset_labels: Optional[List[str]] = None`.
The description mentions it's optional, but the way it is used in line 71 suggests that it is not.

Reply:
Never mind the remark about line 71, I see that it is read from the data frame instead, after apply_subsets.